### ISSUE TITLE
PKGBUILD patch required for OpenSSL 1.1.0

### ISSUE THREAD

**Issue by MSP-Greg** [MSP-Greg/ruby-makepkg-mingw#1]
@larskanis

The dll file names have changed in OpenSSL 1.1.0*.  Hence, the file names need to be added to code in [`ext/openssl/extconf.rb`](https://github.com/ruby/ruby/blob/trunk/ext/openssl/extconf.rb).

Hence, a patch file needs to be added to `PKGBUILD`.  See the bottom of the [README](https://github.com/MSP-Greg/ruby-makepkg-mingw) for code and SHA256.

The patch file is [0003-new-openssl-filenames.patch](https://github.com/MSP-Greg/ruby-makepkg-mingw/blob/master/ruby_build_patches/0003-new-openssl-filenames.patch).

@rhenium I assume this change  might be added to ruby?  I could do the PR, but I'd like someone a little more familiar with build files to look at it.  I'm currently running on Ruby 2.5 & OpenSSL 1.1.0f generated with the patch, and I generate [Ruby, Rails, AWS docs](https://msp-greg.github.io/) with it (code hits https://GitHub.com)...

**Comment by rhenium** [comment]
Hmm, ${prefix}/lib/libcrypto.dll.a and ${prefix}/lib/libssl.dll.a still exist in OpenSSL 1.1.x. Doesn't ext/openssl/extconf.rb find them?

**Comment by MSP-Greg** [comment]
@rhenium Sorry, I should have had line numbers in the above link.

See [`ext/openssl/extconf.rb`](https://github.com/ruby/ruby/blob/trunk/ext/openssl/extconf.rb#L47-L48)

It might have been rather late when I was having the problem.  In my mingw package, I have
```
bin/libcrypto-1_1-x64.dll
bin/libssl-1_1-x64.dll

lib/libcrypto.a
lib/libcrypto.dll.a
lib/libssl.a
lib/libssl.dll.a
```
In the patch file link above, I added `libcrypto-1_1` to the array used in the test for `CRYPTO_malloc`, and `libssl-1_1` to the array used in the test for `SSL_new`.  I believe that solved the problem.  Since `lib` is used with `libeay32`, maybe the arrays just need `crypto` changed to `libcrypto` and same for `ssl`.

I had previously used 1.1.0c, but the patch file on my system is dated after 1.1.0e was released.  Maybe OpenSSL decided to prefix the names with `lib`, since some systems look for that...  I don't know.

Re my question in [ruby / openssl](https://github.com/ruby/openssl/issues/113), I've got ruby / ruby forked, but not ruby / openssl.

Lastly, if there's anything you'd like me to check, I'd be happy to.

**Comment by rhenium** [comment]
What `have_library("xxx")` basically does is to check whether if `gcc ... -lxxx ...` runs successfully. This will find `libxxx.dll.a` as well as `xxx.dll.a` according to the documentation of GNU ld[1], and actually finds on my machine. Anyways I don't think `have_library("libcrypto-1_1")` will help finding `libcrypto-1_1-x64.dll`...

[1] https://sourceware.org/git/?p=binutils.git;a=blob;f=ld/ld.texinfo;h=e4788f67b57fd9866148dcb37969a2d6fd7e9c75;hb=HEAD#l7288

**Comment by MSP-Greg** [comment]
@rhenium Your're correct.

Ran the following from the msys2/mingw shell --
```
require 'mkmf'
have_library("crypto"       , "CRYPTO_malloc")  #=> yes
have_library("libcrypto"    , "CRYPTO_malloc")  #=> yes
have_library("libcrypto-1_1", "CRYPTO_malloc")  #=> no
```

Hence, you're correct.

I'll remove the patch.  As I mentioned, I was having some issues moving from Rubyinstaller to makepkg-mingw / PKGBUILD, and I was having inconsistent builds and tests. `make -j1` solved them.  I know I did see the error message, as I would have never ended up in that file...

I'll do a build (US) tonite without the patch.

Sorry, and thanks for help.
